### PR TITLE
Fixes #3993

### LINKED ISSUE #3993: dup fails for embedded documents that use store_as without using Mongoid::Attributes::Dynamic
```ruby
class Test1
  include Mongoid::Document
  embeds_one :test2, :store_as => :t
  field :name
end

class Test2
  include Mongoid::Document
  embedded_in :test1
  field :name
end

t = Test1.new(:name => "hi")
t.build_test2(:name => "there")
t.save

# Will throw an error
t.dup()

# Mongoid::Errors::UnknownAttribute:
# Problem:
#  Attempted to set a value for 't' which is not allowed on the model Test1.
# Summary:
#  Without including Mongoid::Attributes::Dynamic in your model and the attribute does not already exist in the attributes hash, attempting to call Test1#t= for it is not allowed. This is also triggered by passing the attribute to any method that accepts an attributes hash, and is raised instead of getting a NoMethodError.
# Resolution:
#  You can include Mongoid::Attributes::Dynamic if you expect to be writing values for undefined fields often.
```

### PR DESCRIPTION
**Pull request by jonhyman**

### CODE REVIEW

**Comment by jonhyman** [comment]
@arthurnn I can't get this test to fail locally, either, but Travis says it failed a spec on one of the builds (`Mongoid::Scopable.default_scope when there is more then one default_scope delays execution of the merge until called`).

**Comment by jonhyman** [comment]
@arthurnn thoughts on merging this in?

File: spec/mongoid/copyable_spec.rb [hunk 1]
```diff
@@ -54,6 +54,30 @@
         end
       end
 
+      context "when using store_as" do
+        context "and dynamic attributes are not set" do
```
Comment (arthurnn) [inline spec/mongoid/copyable_spec.rb]: can you add space here and in the next line, to keep the file style consistent ?

File: spec/mongoid/copyable_spec.rb [hunk 2]
```diff
@@ -54,6 +54,30 @@
...
+            class StoreAsDupTest2
+              include Mongoid::Document
+              embedded_in :store_as_dup_test1
+              field :name
+            end
+
+            t = StoreAsDupTest1.new(:name => "hi")
+            t.build_store_as_dup_test2(:name => "there")
+            t.save
+            copy = t.__send__(method)
```
Comment (arthurnn) [inline spec/mongoid/copyable_spec.rb]: can you use a normal `send` call here? instead of `__send__`

File: spec/mongoid/copyable_spec.rb [hunk 3]
```diff
@@ -54,6 +54,30 @@
         end
       end
 
+      context "when using store_as" do
+        context "and dynamic attributes are not set" do
+          it "clones" do
+            class StoreAsDupTest1
```
Comment (arthurnn) [inline spec/mongoid/copyable_spec.rb]: I think it is better to define this classes outside this test, as this is in a loop.

**Comment by arthurnn** [comment]
Can you also amend commit a changelog entry for this bug fix?
thanks

File: spec/mongoid/copyable_spec.rb (hunk 1 above)
Comment (jonhyman) [inline spec/mongoid/copyable_spec.rb]: Sure

File: spec/mongoid/copyable_spec.rb (hunk 3 above)
Comment (jonhyman) [inline spec/mongoid/copyable_spec.rb]: Will do, I'll move to their own files.

File: spec/mongoid/copyable_spec.rb (hunk 2 above)
Comment (jonhyman) [inline spec/mongoid/copyable_spec.rb]: Yep.

**Comment by jonhyman** [comment]
@arthurnn please check now and let me know your comments. Thanks!

File: CHANGELOG.md [hunk 4]
```diff
@@ -11,6 +11,10 @@ For instructions on upgrading to newer versions, visit
 
 * \#3985 Return nil when using `{upsert: true}` in `find_and_modify` (Adrien Siami)
 
+### Resolved Issues
+
+* \##3993 dup/clone fails for embedded documents that use store_as without using Mongoid::Atributes::Dynamic
```
Comment (arthurnn) [inline CHANGELOG.md]: There is an extra `#` .

The changelog, probably should say:
>  Fix an issue where `dup/clone` fails....

Comment (jonhyman) [inline CHANGELOG.md]: Done

**Comment by jonhyman** [comment]
@arthurnn can this be merged?

**Comment by arthurnn** [comment]
thanks

**Pull request merged**